test/e2e/framework.go

```diff
@@ -1986,6 +1986,9 @@ func (data *TestData) CreateServiceWithAnnotations(serviceName, namespace string
 	if (serviceType == corev1.ServiceTypeNodePort || serviceType == corev1.ServiceTypeLoadBalancer) && nodeLocalExternal {
 		service.Spec.ExternalTrafficPolicy = corev1.ServiceExternalTrafficPolicyTypeLocal
 	}
+	for _, mutator := range mutators {
+		mutator(&service)
+	}
```

Comment:
Idea for a future PR: it may be convenient to have a `ServiceBuilder` for tests, like we have a `PodBuilder`

Reply:
Sure, I could follow up with a PR.